analyses/mutational-signatures/07-plot_cns_fit.Rmd

```diff
@@ -0,0 +1,485 @@
+  ) +
+  ggpubr::theme_pubr() + 
+  cowplot::panel_border() +
+  ggplot2::theme(
+    # angle needed to fit "Other" category
+    axis.text.x = ggplot2::element_text(size = 7.5, angle = 45, hjust = 1),
+    axis.text.y = ggplot2::element_text(size = 9),
+    axis.title = ggplot2::element_text(size = 12),
+    strip.text = ggplot2::element_text(size = 8.5),
+    legend.position = "none"
+  )
+
+exposures_sina_boxplot
+
+ggplot2::ggsave(exposures_sina_IQR_file, exposures_sina_boxplot, width = 5, height = 8)
+
+```
+
+
+### Barplots of proportions of samples that are _unexposed_ 
+
+Notably, the "rug" of 0's in each distribution represents samples which are unexposed to the given signature. 
+It's useful to get a clear sense of the proportion of samples that are totally unexposed.
+We do _not_ include "Other" in this plot. 
+
+```{r proportion, fig.width = 5, fig.height = 10}
+exposures_proportions <- results_df %>%
+  # remove "other" - not useful to know if samples are unexposed to signatures that aren't even present.
+  dplyr::filter(signature != "Other") %>%
+  dplyr::group_by(cancer_group_display_n, cancer_group_hex, signature) %>%
+  dplyr::summarize(percent_exposed = sum(exposure > 0)/dplyr::n()) %>%
+  ggplot2::ggplot() + 
+  ggplot2::aes(x = signature, y = percent_exposed, fill = cancer_group_hex) + 
+  ggplot2::geom_col(color = "black") +
+  ggplot2::facet_wrap(~cancer_group_display_n, nrow = 5) + 
+  ggplot2::scale_fill_identity() +
+  ggplot2::labs(x = "RefSig Signature", 
+                y = "Percent of samples exposed to signature") +
+  ggpubr::theme_pubr() +
+  ggplot2::theme(axis.text.x = ggplot2::element_text(size = 8))
+
+exposures_proportions
+
+ggplot2::ggsave(exposures_presence_barplot_file, exposures_proportions, width = 5, height = 10)
+
+```
+
+
+<br><br>
+
+
+### Barplot of sample exposures faceted across cancer groups
+
+This plot shows the literal exposures for each samples.
+We order plots by sample 1, and we include the unknown proportion so everything has the same height. 
+In this case, "Unknown" specifically means mutations that were not assigned to the specific 8 CNS signatures examined, but are associated (if at all) with a different signature.
+
+```{r barplot1, fig.width = 12, fig.height = 9}
+
+# Make a column for exposure of interest to order samples by
+samples_in_order <- results_df %>%
+  dplyr::filter(signature == 1) %>%
+  dplyr::select(exposure, Kids_First_Biospecimen_ID) %>%
+  dplyr::arrange(-exposure) %>%
+  dplyr::pull(Kids_First_Biospecimen_ID)
+
+
+sample_exposures_barplot <- results_df %>%
+  dplyr::mutate(Kids_First_Biospecimen_ID = factor(Kids_First_Biospecimen_ID, levels = samples_in_order)) %>%
+  ggplot2::ggplot() +
+  ggplot2::aes(x = Kids_First_Biospecimen_ID, 
+               y = exposure, 
+               fill = signature) + 
+  ggplot2::geom_col(color = "black", size = 0.05) + 
+  ggplot2::labs(x = "Sample", y = "Signature exposure", fill = "RefSig Signature") + 
```

Comment:
This y-axis label appears to be using "`sigfit` parlance" – I wonder if that's potentially confusing to an audience that's familiar with `deconstructSigs`? I might be overthinking it.

Reply:
Do you mean the term "exposure," vs something like "weights"?

Reply:
Noting I ended up changing labels with "exposure" to use the word "weights"